odbc/src/api/functions.rs

```diff
@@ -1966,23 +2013,62 @@ pub unsafe extern "C" fn SQLSetConnectAttr(
 /// This is the WChar version of the SQLSetConnectAttr function
 ///
 /// # Safety
-/// Because this is a C-infereface, this is necessarily unsafe
+/// Because this is a C-interface, this is necessarily unsafe
 ///
 #[no_mangle]
 pub unsafe extern "C" fn SQLSetConnectAttrW(
-    _hdbc: HDbc,
-    _attr: ConnectionAttribute,
-    _value: Pointer,
-    _str_length: Integer,
+    connection_handle: HDbc,
+    attribute: ConnectionAttribute,
+    value_ptr: Pointer,
+    str_length: Integer,
 ) -> SqlReturn {
-    unimplemented!()
+    let mut err = None;
+    let conn_handle = MongoHandleRef::from(connection_handle);
+
+    // This scope is introduced to make the RWLock Guard expire before we write
+    // any error values via add_diag_info as RWLock::write is not reentrant on
+    // all operating systems, and the docs say it can panic.
+    let sql_return = {
+        let conn = must_be_valid!((*conn_handle).as_connection());
+        let mut conn_guard = conn.write().unwrap();
+
+        match attribute {
+            ConnectionAttribute::CurrentCatalog => {
+                let current_catalog =
+                    input_wtext_to_string(value_ptr as *const WChar, str_length as usize);
+                conn_guard.attributes.current_catalog = Some(current_catalog);
```

Comment:
If the catalog name is a not a database the user has access to, then this should return SQL_ERROR and 3D000-Invalid catalog name.

Reply:
discussed offline -- deferring this work until after the epic since it is not required to get PowerBI working. Followup ticket: https://jira.mongodb.org/browse/SQL-1075